[docs] Document PostHog functions for custom builds (#47760)
expo/expo: docs/pages/custom-builds/schema.mdx

  href="https://github.com/expo/eas-cli/blob/main/packages/build-tools/src/steps/functions/sendSlackMessage.ts"
/>

#### `eas/posthog_capture_event`

> Sends an analytics event to [PostHog](https://posthog.com/). Use it to mark builds, releases, and other milestones on your PostHog timeline. Run `eas integrations:posthog:connect` to link a PostHog project and set the environment variables these functions read. `eas/posthog_capture_event` uses your public project API key; the other PostHog functions use a personal API key with the scope noted for each one. When you do not provide a `distinct_id`, the event is sent anonymously and does not create a PostHog [person profile](https://posthog.com/docs/data/persons).

Reviewer: **[suggestion]** Two thoughts joined by a semicolon; the style guide asks for one thought per sentence (`style-guide.md#voice-and-tone`).

```suggestion
Sends an analytics event to [PostHog](https://posthog.com/). Use it to mark builds, releases, and other milestones on your PostHog timeline. Run `eas integrations:posthog:connect` to link a PostHog project and set the environment variables these functions read. `eas/posthog_capture_event` uses your public project API key. The other PostHog functions use a personal API key with the scope noted for each one. When you do not provide a `distinct_id`, the event is sent anonymously and does not create a PostHog [person profile](https://posthog.com/docs/data/persons).
```

Author: applied, thank you!

          flag: new-checkout
          rollout_percentage: 25
```

| Input                | Type      | Description                                                                                                                                                                                                                                         |
| -------------------- | --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `flag`               | `string`  | Required. The key of the feature flag to update.                                                                                                                                                                                                    |
| `active`             | `boolean` | Whether the flag is enabled.                                                                                                                                                                                                                        |

> | `rollout_percentage` | `number`  | Percentage of users the flag rolls out to, as an integer from `0` to `100`. It is applied to the flag's catch-all release condition and preserves other conditions. When the flag has no catch-all condition, it is applied to the first condition. |

Reviewer: **[suggestion]** "It is applied" twice is passive voice with an unclear subject; active names the actor (`style-guide.md#voice-and-tone`).

```suggestion
| `rollout_percentage` | `number`  | Percentage of users the flag rolls out to, as an integer from `0` to `100`. The function applies it to the flag's catch-all release condition and preserves other conditions. When the flag has no catch-all condition, the function applies it to the first condition. |
```

Author: applied, same fix on the workflows syntax page too

| `ignore_error`       | `boolean` | When `true`, a network error, a missing flag, or an unexpected response is logged but does not fail the step. Defaults to `false`. A permissions error, or invalid input such as an out-of-range `rollout_percentage`, always fails the step.       |

<BoxLink
  title="eas/posthog_flag_rollout source code"
  description="View the source code for the eas/posthog_flag_rollout function on GitHub."
  Icon={GithubIcon}
  href="https://github.com/expo/eas-cli/blob/main/packages/build-tools/src/steps/functions/rolloutPosthogFlag.ts"
/>

#### `eas/posthog_wait_for_metric`

Pauses the build until a [HogQL](https://posthog.com/docs/hogql) query returns a number that satisfies a comparison. Use it to gate on a metric, such as holding until the error count over the last few minutes stays low. The function runs the query every `interval_seconds` until the comparison is true or `timeout_seconds` elapses.

> > **info** This step has no `ignore_error` input. A gate you ignore is not a gate, so a timeout or an unreadable query always fails the step.

Reviewer: **[suggestion]** "A gate you ignore is not a gate" is a rhetorical flourish; callouts read best neutral and direct (`style-guide.md#voice-and-tone`). The sibling `eas/posthog_wait_for_query` callout already uses the plain phrasing.

```suggestion
> **info** This step has no `ignore_error` input. A timeout or an unreadable query always fails the step.
```

Author: fair, that one was me having fun 😄 went with your plain phrasing, and removed the same flourish from the syntax page for consistency 🙇‍♂️ Ty!

| `date_marker`  | `string`  | The ISO 8601 timestamp the annotation is pinned to. Defaults to the current time.                                                                            |
| `api_key`      | `string`  | PostHog personal API key. Defaults to the `POSTHOG_CLI_API_KEY` environment variable. Requires the `annotation:write` scope.                                 |
| `project_id`   | `string`  | PostHog project ID. Defaults to the `POSTHOG_CLI_PROJECT_ID` environment variable.                                                                           |
| `ignore_error` | `boolean` | When `true`, a network error or an unexpected response is logged but does not fail the step. Defaults to `false`. A permissions error always fails the step. |

<BoxLink
  title="eas/posthog_annotation source code"
  description="View the source code for the eas/posthog_annotation function on GitHub."
  Icon={GithubIcon}
  href="https://github.com/expo/eas-cli/blob/main/packages/build-tools/src/steps/functions/createPosthogAnnotation.ts"
/>

#### `eas/posthog_upload_sourcemaps`

> Uploads JavaScript source maps to PostHog so that stack traces in [error tracking](/guides/using-posthog/#error-tracking) are symbolicated. Run it after the step that produces your bundle, in the same job, so the bundle and source maps are on disk. Configure the PostHog Metro config from the [Source maps guide](/guides/using-posthog/#source-maps) so bundles carry the chunk IDs that match them to their source maps.

Reviewer: **[suggestion]** Passive "are symbolicated" where active works (`style-guide.md#voice-and-tone`).

```suggestion
Uploads JavaScript source maps to PostHog so that PostHog symbolicates stack traces in [error tracking](/guides/using-posthog/#error-tracking). Run it after the step that produces your bundle, in the same job, so the bundle and source maps are on disk. Configure the PostHog Metro config from the [Source maps guide](/guides/using-posthog/#source-maps) so bundles carry the chunk IDs that match them to their source maps.
```

Author: applied, mirrored on the syntax page as well 👍
